ProcessLib/ThermoMechanicalPhaseField/ThermoMechanicalPhaseFieldFEM-impl.h

```diff
@@ -0,0 +1,649 @@
+        Kud;
+    Kud.setZero(displacement_size, phasefield_size);
+
+    typename ShapeMatricesType::template MatrixType<displacement_size,
+                                                    temperature_size>
+        KuT;
+    KuT.setZero(displacement_size, temperature_size);
+
+    typename ShapeMatricesType::template MatrixType<phasefield_size,
+                                                    displacement_size>
+        Kdu;
+    Kdu.setZero(phasefield_size, displacement_size);
+
+    typename ShapeMatricesType::template MatrixType<phasefield_size,
+                                                    temperature_size>
+        KdT;
+    KdT.setZero(phasefield_size, temperature_size);
+
+    typename ShapeMatricesType::template MatrixType<temperature_size,
+                                                    phasefield_size>
+        KTd;
+    KTd.setZero(temperature_size, phasefield_size);
+
+    typename ShapeMatricesType::NodalMatrixType KTT;
+    KTT.setZero(temperature_size, temperature_size);
+
+    typename ShapeMatricesType::NodalMatrixType DTT;
+    DTT.setZero(temperature_size, temperature_size);
+
+    typename ShapeMatricesType::NodalMatrixType Kdd;
+    Kdd.setZero(phasefield_size, phasefield_size);
+
+    typename ShapeMatricesType::NodalMatrixType Ddd;
+    Ddd.setZero(phasefield_size, phasefield_size);
+
+    double const& dt = _process_data.dt;
+
+    SpatialPosition x_position;
+    x_position.setElementID(_element.getID());
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        x_position.setIntegrationPoint(ip);
+        auto const& w = _ip_data[ip].integration_weight;
+
+        auto const& dNdx = _ip_data[ip].dNdx;
+        auto const& N = _ip_data[ip].N;
+
+        auto const x_coord =
+            interpolateXCoordinate<ShapeFunction, ShapeMatricesType>(_element,
+                                                                     N);
+        auto const& B =
+            LinearBMatrix::computeBMatrix<DisplacementDim,
+                                          ShapeFunction::NPOINTS,
+                                          typename BMatricesType::BMatrixType>(
+                dNdx, N, x_coord, _is_axially_symmetric);
+
+        auto& eps_m = _ip_data[ip].eps_m;
+        auto& eps = _ip_data[ip].eps;
+
+        auto const& C_tensile = _ip_data[ip].C_tensile;
+        auto const& C_compressive = _ip_data[ip].C_compressive;
+
+        auto const& strain_energy_tensile = _ip_data[ip].strain_energy_tensile;
+        auto const& sigma_tensile = _ip_data[ip].sigma_tensile;
+
+        auto& history_variable = _ip_data[ip].history_variable;
+        auto& history_variable_prev = _ip_data[ip].history_variable_prev;
+
+        auto const& sigma_real = _ip_data[ip].sigma_real;
+
+        // auto const [&](member){ return _process_data.member(t,
```

Comment:
Please remove all commented code parts or add a debugging description if something is needed for such a case.